### PR TITLE
Fixing validation of Cors hostname string

### PR DESCRIPTION
**Pull request by cjbooms**
Signed-off-by: Conor Gallagher <conor.gallagher@zalando.ie>

### CODE REVIEW

**Comment by cjbooms** [comment]
👍

**Comment by irishshagua** [comment]
Be good to throw a unit test in as well...

**Comment by cjbooms** [comment]
There are tests for Cors already. I think it's fine without. We'd just be testing the operation of a library function

**Comment by irishshagua** [comment]
> There are tests for Cors already. I think it's fine without. We'd just be testing the operation of a library function

If it's fixing a bug there should be a test. I pushed a small one there...

**Comment by irishshagua** [comment]
:+1:

**Comment by cjbooms** [comment]
👍

**Comment by cjbooms** [comment]
👍

File: src/main/scala/ie/zalando/fabric/gateway/service/ResourcePersistenceValidations.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ object ResourcePersistenceValidations {
       validationRequest.resource.paths.keys.toList.map(validateStarStarPathPosition).sequence,
       validateStackSetIntegration(validationRequest.hasExternallyManagedServices, validationRequest.definedServiceCount),
       validationRequest.corsConfig
-        .map(_.allowedOrigins.toList.map(hostname => validateCorsHostname(hostname)))
+        .map(_.allowedOrigins.toList.map(hostname => validateCorsUri(hostname)))
```
Comment (herojan) [inline src/main/scala/ie/zalando/fabric/gateway/service/ResourcePersistenceValidations.scala:67]: I'd say naming-wise "hostname" should be "origin" and "validateCorsUri" should be "validateCorsOrigin"

**Comment by cjbooms** [comment]
👍

**Comment by herojan** [comment]
:+1:

**Pull request merged**